lib/Conversion/TorchToLinalg/Reduction.cpp

```diff
@@ -322,6 +389,46 @@ class ConvertReductionOp : public ConversionPattern {
     auto resultType = getTypeConverter()
                           ->convertType(op->getResult(0).getType())
                           .cast<RankedTensorType>();
+
+    if (auto normOp = dyn_cast<AtenLinalgVectorNormOp>(op)) {
+      auto elemType = resultType.getElementType();
+
+      if (!elemType.isa<mlir::FloatType>())
+        return rewriter.notifyMatchFailure(
+            op, "only float types are valid for vector norm ops");
+
+      // Cast `ord` to float so that we can readily pass it math.powf.
+      auto ordValue = operands[1];
+      if (ordValue.getType().isa<mlir::IntegerType>())
+        ordValue = rewriter.create<arith::SIToFPOp>(loc, elemType, ordValue);
+
+      // Sum each element of the tensor after computing the exponentiation.
+      auto sumOp =
+          createNormSumReduction(loc, elemType, opInfo, ordValue, rewriter);
+      if (!sumOp)
+        return failure();
```

Comment:
I know this practice hasn't been followed that much in this file, but can you use `notifyMatchFailure` to give an error message every time a failure happens?

Reply:
Fixed in commit de84ccc.